File: app/src/main/java/dev/xuanran/codebook/util/FileUtils.java
package dev.xuanran.codebook.util;

import android.content.Context;
import android.content.res.AssetManager;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class FileUtils {
    public static String readAssetTextFile(Context context, String fileName) {
        StringBuilder text = new StringBuilder();
        AssetManager assetManager = context.getAssets();

        try (InputStream inputStream = assetManager.open(fileName);
             BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
            String line;
            while ((line = reader.readLine()) != null) {
                text.append(line).append("\n");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }

        return text.toString();
    }
}

File: app/src/main/java/dev/xuanran/codebook/util/FileUtils.kt
package dev.xuanran.codebook.util

import android.content.Context
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import java.lang.StringBuilder

object FileUtils {
    fun readAssetTextFile(context: Context, fileName: String): String {
        val text = StringBuilder()
        val assetManager = context.getAssets()

        try {
            assetManager.open(fileName).use { inputStream ->
                BufferedReader(InputStreamReader(inputStream)).use { reader ->
                    var line: String?
                    while ((reader.readLine().also { line = it }) != null) {
                        text.append(line).append("\n")
                    }
                }
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return text.toString()
    }
}
